src/Nethermind/Nethermind.JsonRpc/Modules/Eth/EthRpcModule.cs

```diff
@@ -85,7 +89,7 @@ private static bool HasStateForBlock(IBlockchainBridge blockchainBridge, BlockHe
 
     public ResultWrapper<string> eth_protocolVersion()
     {
-        int highestVersion = P2PProtocolInfoProvider.GetHighestVersionOfEthProtocol();
+        int highestVersion = _protocolsManager.GetHighestProtocolVersion(Protocol.Eth);
```

Comment:
You know, you can extract this method out into its own class, let say `IEthP2PRpcModule`, register it as another module and skip a whole lot of change.

Reply:
Which method exactly and how would that help?

Reply:
Just `eth_protocolVersion` since that is the only one that use `_protocolsManager`. Eg: Make an `IEthCommonRpcModule`, put `eth_protocolVersion` into it, moving it out of `IEthRpcModule`. If no other plugin is using it, then you don't need to change other plugin.